Firestore/core/test/unit/local/local_serializer_test.cc

```diff
@@ -438,21 +438,36 @@ TEST_F(LocalSerializerTest, MultipleMutationsAreSquashed) {
   auto message = Message<firestore_client_WriteBatch>::TryParse(&reader);
   MutationBatch decoded = serializer.DecodeMutationBatch(&reader, *message);
   ASSERT_EQ(5, decoded.mutations().size());
-  _google_firestore_v1_Write encoded =
-      remote_serializer.EncodeMutation(decoded.mutations()[0]);
-  ExpectSet(encoded);
-  ExpectNoUpdateTransform(encoded);
-  encoded = remote_serializer.EncodeMutation(decoded.mutations()[1]);
-  ExpectSet(encoded);
-  ExpectUpdateTransform(encoded);
-  encoded = remote_serializer.EncodeMutation(decoded.mutations()[2]);
-  ExpectDelete(encoded);
-  encoded = remote_serializer.EncodeMutation(decoded.mutations()[3]);
-  ExpectPatch(encoded);
-  ExpectUpdateTransform(encoded);
-  encoded = remote_serializer.EncodeMutation(decoded.mutations()[4]);
-  ExpectPatch(encoded);
-  ExpectNoUpdateTransform(encoded);
+
+  {
+    Message<google_firestore_v1_Write> encoded{
+        remote_serializer.EncodeMutation(decoded.mutations()[0])};
+    ExpectSet(*encoded);
+    ExpectNoUpdateTransform(*encoded);
+  }
+  {
+    Message<google_firestore_v1_Write> encoded{
```

Comment:
Right, we're trying to assign `T` but the assignment operator expects a `Message<T>`. There's a couple of alternatives:
- sorry, I don't remember if we have `MakeMessage` or only `MakeSharedMessage` -- in the latter case, `MakeMessage` should be easy to define. With `MakeMessage`, this could be `encoded = MakeMessage(/*argument*/);`;
- we could add a `reset` member function to `Message`, similar in spirit to `std::unique_ptr::reset` and taking `const T&` as the argument, like the constructor;
- leave as is (this is test code, after all).

What do you think?

Reply:
I must have been very tired when I tried this. Using `MakeMessage` obviously works here.